zk-token-sdk/src/sigma_proofs/mod.rs

```diff
@@ -38,7 +38,9 @@ fn ristretto_point_from_optional_slice(
 ) -> Result<CompressedRistretto, SigmaProofVerificationError> {
     optional_slice
         .and_then(|slice| (slice.len() == RISTRETTO_POINT_LEN).then_some(slice))
-        .map(CompressedRistretto::from_slice)
+        .map(|slice| {
+            CompressedRistretto::from_slice(slice).expect("Input slice should have a length of 32")
+        })
```

Comment:
```suggestion
        .and_then(|slice| CompressedRistretto::from_slice(slice).ok())
```
Since we are returning `None` if the input is wrong length, it seems like we can just add `.ok()` here and remove the line above that does the length check.